lib/Common/Core/CRC.cpp

```diff
@@ -25,3 +25,31 @@ unsigned int CalculateCRC32(const char* in)
     }
     return crc ^ (unsigned int)-1;
 }
+
+uint CalculateCRC(uint bufferCRC, size_t data)
```

Comment:
It's not super clear from the names that CalculateCRC is the same as CalculateCRC32 except that it will try to use built-in instructions when appropriate. Maybe the other CalculateCRC32 could become CalculateCRC32Impl, or even move into an anonymous namespace so nobody accidentally calls the slow version?